vpr/src/route/rr_graph.cpp

```diff
@@ -94,8 +94,25 @@ std::set<int> get_layers_pin_is_connected_to(const t_physical_tile_type_ptr type
 ///@brief given a specific layer number and type, it returns layers which have same pin_index connected to the given layer
 std::set<int> get_layers_connected_to_pin(const t_physical_tile_type_ptr type, int to_layer, int pin_index);
 
+///@brief checks whether the channel width has been changed or not
 bool channel_widths_unchanged(const t_chan_width& current, const t_chan_width& proposed);
 
+/**
+ * @brief This routine calculate pin connections to tracks for a specific type on the Fc value defined for each pin in the architecture file.
```

Comment:
What is the difference between this routine and the one below it?  From their comments they both seem to do the same thing. One returns a 5D matrix of vectors, and the other a 6D matrix, but other than that they don't seem different.  Could one be deleted (don't worry about it if that is too much work, but we could put a TODO in if that's the better long term path).

Reply:
The one that returns the 6D matrix is written as a helper function for the other one. Basically, we first calculate the number of connections should be made for each pin for the current tile and save a relative track indices. Then, we return the 6D matrix and post process it to calculate the absolute track number and keep a vector of connection per each pin. Note that basically they return the same thing (6D matrix of int and 5D matrix of integer vector) but the logic within each function are different.